core/src/main/scala/sttp/client3/internal/package.scala

```diff
@@ -30,6 +34,18 @@ package object internal {
     transfer()
   }
 
+  private[client3] def withCharset(headers: Map[String, String], encoding: String): Map[String, String] = {
+    val contentTypeOpt = for {
+      charset <- Try(Charset.forName(encoding)).toOption
```

Comment:
do we need this validation? if the user wants to specify an invalid charset, I'd let them do so. Or throw an exception, but that might be too restrictive

Reply:
but the user specifies the encoding, and we are changing it to the charset behind the scenes, so it's maybe better to keep content-type untouched if the encoding is incorrect?

Reply:
I think silent errors might be very confusing

Reply:
ok, so i will get rid of it